api/src/Messages/ApiResource/MessageUnread.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace App\Messages\ApiResource;
+
+use ApiPlatform\Metadata\ApiResource;
+use ApiPlatform\Metadata\Get;
+use ApiPlatform\Metadata\Link;
+use App\Entity\Discussion;
+use App\Entity\Repairer;
+use App\Messages\State\NumberOfMessageNotReadForCustomerProvider;
+use App\Messages\State\NumberOfMessageNotReadForDiscussionProvider;
+use App\Messages\State\NumberOfMessageNotReadForRepairerProvider;
+
+#[ApiResource]
+#[Get(
+    uriTemplate: '/customers/messages_unread',
+    security: 'is_granted("IS_AUTHENTICATED_FULLY")',
+    provider: NumberOfMessageNotReadForCustomerProvider::class,
+)]
+#[Get(
+    uriTemplate: '/repairers/{repairer_id}/messages_unread',
+    uriVariables: [
+        'repairer_id' => new Link(
+            toProperty: 'repairer',
+            fromClass: Repairer::class,
+        ),
+    ],
+    security: 'is_granted("IS_AUTHENTICATED_FULLY") and user.isAssociatedWithRepairer(repairer_id)',
+    provider: NumberOfMessageNotReadForRepairerProvider::class,
+)]
+#[Get(
+    uriTemplate: '/messages_unread/{discussion_id}',
+    uriVariables: [
+        'discussion_id' => new Link(
+            fromProperty: 'id',
+            fromClass: Discussion::class,
+        ),
+    ],
+    security: 'is_granted("MESSAGE_UNREAD_BY_DISCUSSION", discussion_id)',
+    provider: NumberOfMessageNotReadForDiscussionProvider::class,
+)]
+class MessageUnread
```

Comment:
```suggestion
final class MessageUnread
```

Same, generally it is good to make a class final. Unless you think it can be extended.